sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -405,6 +411,7 @@ public partial class RetryOptions
         public Azure.Core.RetryMode Mode { get { throw null; } set { } }
         public System.TimeSpan NetworkTimeout { get { throw null; } set { } }
     }
+    public delegate System.Threading.Tasks.Task SyncAsyncEventHandler<T>(T e) where T : Azure.SyncAsyncEventArgs;
```

Comment:
no TClient?

Reply:
@KrzysztofCwalina wants to put that in the `EventArgs`.  If you look at the Search APIs in particular you'll see they have an instance of the `SearchIndexingBufferedSender`.

Reply:
Should `Azure.SyncAsyncEventArgs` have `TSender Sender` read only property?  Or is it missing to leave a room for cases where sender should not be exposed (if such exist).

Reply:
I think you'll always need to derive from `SyncAsyncEventArgs` if you're doing anything interesting enough to need a client, at which point you might as well add it yourself with a better name (like `public QueueClient Queue { get; }` for your use case).  I did use `Sender` for the Search examples, but that's only because the relevant type is actually a `SearchIndexingBufferedSender` rather than a `SearchClient`.

I think the only reason it'd be worth adding a generic `TSender` is if we think we'll have a lot of events with no extra data.  I'd be surprised if we end up with any of those.